apps/omg_watcher/test/omg_watcher/integration/in_flight_exit_test.exs

```diff
@@ -257,7 +257,82 @@ defmodule OMG.Watcher.Integration.InFlightExitTest do
            end) =~ "Invalid in-flight exit finalization"
 
     assert %{"in_flight_exits" => [_], "byzantine_events" => byzantine_events} = WatcherHelper.success?("/status.get")
-    assert [%{"event" => "invalid_piggyback"}] = Enum.filter(byzantine_events, &(&1["event"] != "piggyback_available"))
+
+    assert [%{"event" => "unchallenged_piggyback"}, %{"event" => "invalid_piggyback"}] =
+             Enum.filter(byzantine_events, &(&1["event"] != "piggyback_available"))
+  end
+
+  @tag fixtures: [:in_beam_watcher, :alice, :bob, :mix_based_child_chain, :token, :alice_deposits]
+  test "invalid piggyback causes unchallenged exit byzantine event when sla margin period passes",
+       %{alice: alice, bob: bob, alice_deposits: {deposit_blknum, _}} do
+    DevHelper.import_unlock_fund(bob)
+
+    tx = OMG.TestHelper.create_signed([{deposit_blknum, 0, 0, alice}], @eth, [{alice, 5}, {bob, 4}])
+    ife1 = tx |> Transaction.Signed.encode() |> WatcherHelper.get_in_flight_exit()
+
+    %{"blknum" => blknum} = tx |> Transaction.Signed.encode() |> WatcherHelper.submit()
+    invalidating_tx = OMG.TestHelper.create_encoded([{blknum, 0, 0, alice}], @eth, [{alice, 4}])
+    %{"blknum" => invalidating_blknum} = WatcherHelper.submit(invalidating_tx)
+    IntegrationTest.wait_for_block_fetch(invalidating_blknum, @timeout)
+
+    {:ok, eth_height} = exit_in_flight_and_wait_for_ife(ife1, alice)
+
+    _ = piggyback(tx, 0, :output, alice)
+
+    IntegrationTest.wait_for_byzantine_event(
+      %OMG.Watcher.Event.InvalidPiggyback{}.name,
+      @timeout
+    )
+
+    exit_processor_sla_margin = Application.fetch_env!(:omg_watcher, :exit_processor_sla_margin)
+    DevHelper.wait_for_root_chain_block(eth_height + exit_processor_sla_margin, @timeout)
+
+    IntegrationTest.wait_for_byzantine_event(
+      %OMG.Watcher.Event.UnchallengedPiggyback{}.name,
+      @timeout
+    )
+  end
+
+  @tag fixtures: [:in_beam_watcher, :alice, :bob, :mix_based_child_chain, :token, :alice_deposits]
```

Comment:
we're stepping away from using fixtures. find another way to test this scenarios

Reply:
I'll try to code it in a cabbage test.
@InoMurko it's a pity I didn't know about that before coding them. Let's have a list ofdon't, things we're stepping away from, like: fixtures, single pipes, etc, so anyone can read it before contributing. I think you are best informed on this stuff. Could you prepare that list?

Reply:
I decided to rely just on unit tests and not code a separate e2e scenario. Those events are already emitted in existing integration and e2e tests. Creating a separate cabbage test just for that would be a lot of hassle.